Sources/SwiftDocCUtilities/ArgumentParsing/Subcommands/Merge.swift

```
@@ -0,0 +1,131 @@
+        private static let catalogPathExtension = "docc"
+        
+        // The file manager used to validate the input and output directories.
+        //
+        // Provided as a static variable to allow for using a different file manager in unit tests.
+        static var _fileManager: FileManagerProtocol = FileManager.default
+        
+        // Note:
+        // The order of the option groups in this file is reflected in the 'docc merge --help' output.
+        
+        // MARK: - Inputs & outputs
+        
+        @OptionGroup(title: "Inputs & outputs")
+        var inputsAndOutputs: InputAndOutputOptions
+        struct InputAndOutputOptions: ParsableArguments {
+            @Argument(
+                help: ArgumentHelp(
+                    "A list of paths to '.\(Merge.archivePathExtension)' documentation archive directories to combine into a combined archive.",
+                    valueName: "archive-path"),
+                transform: URL.init(fileURLWithPath:))
+            var archives: [URL]
+            
+            @Option(
+                help: ArgumentHelp(
+                    "Path to a '.\(Merge.catalogPathExtension)' documentation catalog directory with content for the landing page.",
+                    valueName: "catalog-path"),
+                transform: URL.init(fileURLWithPath:))
+            var landingPageCatalog: URL?
+            
+            @Option(
+                name: [.customLong("output-path"), .customShort("o")],
+                help: "The location where the documentation compiler writes the combined documentation archive.",
+                transform: URL.init(fileURLWithPath:)
+            )
+            var providedOutputURL: URL?
+            
+            var outputURL: URL!
+            
+            mutating func validate() throws {
+                let fileManager = Docc.Merge._fileManager
+                
+                guard !archives.isEmpty else {
```

Does it make sense to merge just one archive? Shouldn't we also raise a validation error when `archives.count == 1` ?

---

Yes, merging a single archive is supported. Doing so will inactivate all the external links in that archive.